### PR TITLE
busctl introspect: accept direction="out" for signals

### PR DESCRIPTION
**Pull request by grawity**
According to the D-Bus spec (v0.29),

> The direction element on `<arg>` may be omitted, in which case it defaults to "in" for method calls and "out" for signals. Signals only allow "out" so while direction **may** be specified, it's pointless.

In particular, Qt5 seems to generate introspection data with this. Therefore we still should accept a 'direction' attribute, even if it's useless.

Closes: #4616

### CODE REVIEW

File: src/libsystemd/sd-bus/busctl-introspect.c [hunk 1]
```diff
@@ -458,6 +459,8 @@ static int parse_xml_node(Context *context, const char *prefix, unsigned n_depth
                                                 } else if (streq(argument_direction, "out")) {
                                                         if (!strextend(&context->member_result, argument_type, NULL))
                                                                 return log_oom();
+                                                } else {
```
Comment (keszybz) [inline src/libsystemd/sd-bus/busctl-introspect.c]: No {}.

File: src/libsystemd/sd-bus/busctl-introspect.c [hunk 2]
```diff
@@ -599,8 +604,12 @@ static int parse_xml_node(Context *context, const char *prefix, unsigned n_depth
...
 
                                 if (argument_type) {
-                                        if (!strextend(&context->member_signature, argument_type, NULL))
-                                                return log_oom();
+                                        if (!argument_direction || streq(argument_direction, "out")) {
+                                                if (!strextend(&context->member_signature, argument_type, NULL))
+                                                        return log_oom();
+                                        } else {
+                                                log_error("Unexpected signal <arg> direction value '%s'.", argument_direction);
+                                        }
```
Comment (keszybz) [inline src/libsystemd/sd-bus/busctl-introspect.c]: Ditto.

File: src/libsystemd/sd-bus/busctl-introspect.c [hunk 3]
```diff
@@ -639,6 +648,21 @@ static int parse_xml_node(Context *context, const char *prefix, unsigned n_depth
 
                         break;
 
+                case STATE_SIGNAL_ARG_DIRECTION:
+
+                        if (t == XML_ATTRIBUTE_VALUE) {
+                                free(argument_direction);
+                                argument_direction = name;
+                                name = NULL;
```
Comment (keszybz) [inline src/libsystemd/sd-bus/busctl-introspect.c]: `free_and_replace(argument_direction, name);`

Comment (grawity) [reply src/libsystemd/sd-bus/busctl-introspect.c]: Direct copypaste of STATE_METHOD_ARG_DIRECTION. Should I change this occurence only, or all 12 of them?

Comment (keszybz) [reply src/libsystemd/sd-bus/busctl-introspect.c]: Ah, I didn't notice that. Maybe change the other 11 as a separate commit?

**Comment by grawity** [comment]
v2

**Comment by keszybz** [comment]
CI is still failing on --help return value, we can ignore that.

LGTM.

**Pull request merged**